nanome_vault/VaultServer.py

```diff
@@ -322,8 +370,24 @@ def start_process(cls, port, ssl_cert, keep_files_days, converter_url):
 
         VaultServer.is_running = False
 
-        if cleanup_thread:
-            cleanup_thread.join()
+        if auth_cleanup_thread:
+            auth_cleanup_thread.join()
+
+        if file_cleanup_thread:
+            file_cleanup_thread.join()
+
+    @staticmethod
+    def auth_cleanup():
+        while VaultServer.is_running:
+            expiry_time = datetime.today() - timedelta(hours=1)
+
+            # check token last accessed time and remove those older than 1 hour
+            for token in list(VaultServer.auth_cache):
+                if VaultServer.auth_cache[token]['access'] < expiry_time:
+                    del VaultServer.auth_cache[token]
+
+            # wait 10 minutes
+            sleep(600)
```

Comment:
This will prevent the thread from joining for up to 10 minutes. You should sleep for a second or less and check with a 10 minutes timer if inner code needs to be executed